### ISSUE TITLE
non-specific conversion brush

### ISSUE THREAD

**Issue by ngreenwald** [vanvalenlab/deepcell-label#91]
Sometimes there is one cell that needs to be expanded significantly into multiple adjoining cells. Having the option for conversion brush to overwrite everything, not just a single target cell, would be useful in this case. Maybe an additional hotkey prompt, like a for all?

**Comment by geneva-miller** [comment]
I'm thinking this would be a new brush-based tool, in which case I think it would just be able to draw over background or existing annotations. I haven't encountered data where I've needed that type of tool, so I haven't implemented it (partly to protect myself and other users from accidentally drawing over existing annotations). How often do you run into these annotation cases? If it happens only rarely I think the overwriting brush will remain a low priority, but if it is a significant speed bump I will prioritize it more.

**Comment by ngreenwald** [comment]
This is a fairly low priority, kind of specific use case. 

I think a higher priority is the non-specific erase brush. I thought I had created an issue for this, but I can't find it. It often happens that I want to take chunks out of a few different neighboring cells to carve out space for a new cell.

**Comment by geneva-miller** [comment]
The non-specific eraser would likely be just the overwriting brush with the eraser turned on. I'm thinking this mode would have text at the bottom of the sidebar, like with the conversion brush, to remind the user that this mode is turned on. Within the mode, you could still use the color picker, adjust the value of the brush, and toggle the eraser like with the default brush mode, although I'm on the fence about triggering the conversion brush from there. I'm thinking a keybind of "b".

In the RGB display with label outlines, all outlines are white. In the current zoom branch, labels are drawn with color on top of a greyscale raw image, and they are highlighted with a white outline (instead of the translucent filling that label outline images get to indicate highlighting). The white outline gets extended to a white outline and a red outline when using the conversion brush, to indicate which label is being overwritten and which one is getting expanded. This has been a great visual cue, but I'm not sure how to bring it over to the RGB display in a way that makes sense. I bring this up because this could also be a good visual cue for the non-specific brush (all outlines turn red when non-specific eraser is turned on, most outlines turn red when the non-specific drawing is turned on to indicate that anything can be overwritten). Do you have thoughts on alternative visual cues for highlighting RGB+label outline images?

**Comment by ngreenwald** [comment]
Okay, so b would be a label-agnostic brush. You could pick a label, and
draw with it, which would affect both empty pixels and pixels assigned to
other labels. Or you could turn on erase in this mode, which would then
likewise non-specifically erase over cells. Is that right?

In terms of visualization, we could reserve red as a caliban color, and not
have users be able to select it as a channel in color? In general it's not
a great one when not at max intensity, as it is the hardest to distinguish
from background. We don't usually use it unless we're looking at 4+
channels at a time.

**Comment by geneva-miller** [comment]
Although the UI elements described in this issue are outdated, this may still be a useful feature for annotation in some cases. Perhaps the brush tool could expose this to users as an option via a submenu.

**Comment by tddough98** [comment]
This can definitely be closed once the overlapping labels work is merged. That will introduce three drawing modes: overlap, exclude, and overwrite. Overwrite will act as a non-specific conversion brush that replaces any and all labels with a new label.
